crates/agentgateway/src/http/ext_proc.rs

```diff
@@ -669,16 +730,23 @@ async fn handle_response_for_response_mutation(
 }
 
 fn req_to_header_map(req: &http::Request) -> Option<proto::HeaderMap> {
+	// Get pseudo header values with fallback
```

Comment:
nit: it seems nicer to make the function return all (4) of the headers, then we don't need to encode semantics about headers here.

In ext_authz we can just do `get_pseudo_headers().into_iter().filter(include_request_headers.contains)` or similar